tests/security/cc/ipsec_client.pm

```diff
@@ -0,0 +1,68 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Run CC 'ipsec' client case
+# Maintainer: Liu Xiaojing <xiaojing.liu@suse.com>
+# Tags: poo#101226
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use audit_test;
+use Utils::Architectures;
+use lockapi;
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'root-console';
+
+    my $server_ip = get_var('SERVER_IP', '10.0.2.101');
+    my $client_ip = get_var('CLIENT_IP', '10.0.2.102');
+
+    # We don't run setup_multimachine in s390x, but we need to know the server and client's
+    # ip address, so we add a known ip to eth0.
+    if (is_s390x) {
+        assert_script_run("ip addr add $client_ip/24 dev eth0");
+    }
+
+    mutex_wait('IPSEC_SERVER_READY');
+
+    assert_script_run("cd $audit_test::test_dir/ipsec_configuration/toe");
+
+    # Setup the ipip tunnel to the IPSec gateway and test it
+    assert_script_run("./ipsec_setup_tunnel_toe.sh start $client_ip $server_ip");
+    assert_script_run('ping -W1 -c1 192.168.100.1');
+
+    # Install IPSec configuration
+    assert_script_run('make install');
+
+    # Test the IPSec connection
+    assert_script_run('ipsec start');
+    assert_script_run('stime=$(date +\'%H:%M:%S\')');
+    assert_script_run('ipsec up ikev2suse');
+
+    # Test the IPSec connection
+    assert_script_run('ping -W1 -c1 192.168.250.1');
+
+    # Search for AUDIT SPD/SAD add records
+    assert_script_run('ausearch -ts $stime | grep --color -e \'MAC_IPSEC_EVENT\' -e \'SPD-add\' -e \'SAD-add\'');
+
+    assert_script_run('stime=$(date +\'%H:%M:%S\')');
+    assert_script_run('ipsec down ikev2suse');
+
+    # Search for AUDIT SPD/SAD delete records
+    assert_script_run('ausearch -ts $stime | grep --color -e \'MAC_IPSEC_EVENT\' -e \'SPD-delete\' -e \'SAD-delete\'');
+
+    # Stop the IPSec service
+    assert_script_run('ipsec stop');
+
+    # Stop the ipip tunnel
+    assert_script_run('./ipsec_setup_tunnel_toe.sh stop');
```

Comment:
Not a big deal, can you please do some tear down on s390x for your new added ip addr upon 'eth0'

Reply:
# ip addr delete 192.168.101.2/24 dev eth0

Reply:
modified. Thx